Feature/article model

anusharanganathan opened the pull request:
This PR is for https://github.com/CottageLabs/willow/issues/138

martyn-w commented on willow/app/inputs/rights_statement_input.rb:
```diff
@@ -21,35 +21,35 @@ def build_components(attribute_name, value, index, options)
-      # out << '  </div>'
+      out << "<div class='row'>"
+      out << "  <div class='col-md-3'>"
+      out << template.label_tag(field_name, 'License statement', required: false)
```
I think it might be 'Licence' not 'License'?

martyn-w replied:
though not sure... I see dublin core use 'license' http://dublincore.org/documents/dcmi-terms/#terms-license

anusharanganathan replied:
American versus UK spelling. I'll go with your recommendation

anusharanganathan commented:
@martyn-w I've made changes to the article model, which hopefully address all the issues you raised. I've given details to the points you raised as a response in the comment.

martyn-w commented:
Hi @anusharanganathan,

thanks for making these changes - I'm afraid I still get an error though if I create a new article but don't set the licence start date (which is not a mandatory field):

![screen shot 2017-07-02 at 17 03 57](https://user-images.githubusercontent.com/86833/27771555-7d5a1d48-5f48-11e7-8ace-0683cdcb8f3b.png)

martyn-w commented:
Also to add, if the licence statement is left blank, the follow error results:

![screen shot 2017-07-02 at 17 20 17](https://user-images.githubusercontent.com/86833/27771635-d49a6156-5f4a-11e7-968d-9361fdcb044d.png)

anusharanganathan commented:
Hi @martyn-w, I am really sorry. That was sloppy of me. I do hope I've done a better job testing my code this time. 

The changes I've made to address the issues you were seeing are in commits https://github.com/CottageLabs/willow/pull/180/commits/ec55fe088a1b1db0702b31eec443c82f110759df and https://github.com/CottageLabs/willow/pull/180/commits/fe9f0ac34df48379b21aa7d6f16cc77f569658f7. Is there a better way to do the check? It doesn't look pretty.

I also fixed another issue with the edit form - Required check to be done only for first entry for the field - commit https://github.com/CottageLabs/willow/pull/180/commits/2c240bb3af8650d4a3059255a33867588a15972f. The add another and delete will follow in a PR after this is out of the way

I moved labels and field hints to he locale file - commit https://github.com/CottageLabs/willow/pull/180/commits/e4cc3d518a54c7c1fe199bc836f1a48a090d222a

martyn-w commented:
Hi @anusharanganathan, this is tested and working, excellent job 👍
